### DISCUSSION TITLE
Ajouter dynamiquement un min et un max

### DISCUSSION THREAD

**Discussion by eparentrq** [MTESSDev/FRW#89] [Annonces]
Il serait bien de pouvoir ajouter dynamiquement un minimum et un maximum dans les validations des composantes.
Exemple : la date maximum ou l'année maximum qu'on peut mettre dans un texte ou une liste puisse être crée dynamiquement. puisqu'on peut aller chercher l'année en cours.

Merci.

**Comment by j0ber** [comment]
Bonjour,
Basé sur quoi? La valeur d'un autre champ? 
Vous avez un exemple concret ce serait plus clair.
Merci

**Reply by eparentrq** [reply]
Déjà c'est une bonne idée d'avoir des champs réactifs à ce qu'il y a dans un autre champs (disons une liste de sélection)
Tout dépend de la sélection dans la 1ère liste de sélection, la 2ème se met à jour selon le choix de la 1ère liste.


Pour ce que je suggère, c'est plus mettre un maximum dans un champ texte pour un chiffre, disons une année. L'utilisateur ne peut pas mettre l'année suivante de celle en cours.

ça pourrait donner 
validations:
  max: [{anneeCourante + 1}]

donc, le formulaire serait à jour peu importe l'année en cours.

**Reply by j0ber** [reply]
Pour la liste de sélection c'est déja possible via l'utilisation des domaines de valeurs. Voir exemple 2 dans [cette page](https://formulaires.it.mtess.gouv.qc.ca/fr/Form/7/P700U/0/N/#p=DomainesValeurs).

Par contre pour l'autre élément ça n'existe pas tel que vous le décrivez, mais je vous reviens sous peu sur ce sujet...

**Reply by j0ber** [reply]
Concernant les validations dynamiques, il est possible de créer des validations personnalisées.
Une page a été ajoutée à cet effet dans notre [guide de configuration des formulaires](https://formulaires.it.mtess.gouv.qc.ca/fr/Form/7/P700U/0/N/#p=validationsCustom).
